scripts/imgtool/image.py

```diff
@@ -623,8 +623,15 @@ def pad_to(self, size):
 
     @staticmethod
     def verify(imgfile, key):
-        with open(imgfile, "rb") as f:
-            b = f.read()
+        ext = os.path.splitext(imgfile)[1][1:].lower()
+        try:
+            if ext == INTEL_HEX_EXT:
+                b = IntelHex(imgfile).tobinstr()
+            else:
+                with open(imgfile, 'rb') as f:
+                    b = f.read()
+        except FileNotFoundError:
+            raise click.UsageError("Input file not found")
```

Comment:
nit: Would it worth to mention the input file that is not found is image file? Might be easier for the user to distinguish whether keyfile or the imgfile path is wrong.

Reply:
good point, I'll add the file name as well